docs/source/errors.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: Errors
+subtitle: Error and status codes returned by the Apollo Router
+description: Error codes and response status codes returned by the Apollo Router
+---
+
+Learn about error codes and HTTP response status codes returned by the Apollo Router.
+
+## Status codes
+
+<PropertyList kind="errCodes">
+<Property name="400" short="Bad request">
+
+A problem when validating GraphQL or parsing an HTTP request.
+
+</Property>
+<Property name="401" short="Unauthorized">
+
+Sent an event to a non-existent subscription or used an invalid hash. Returned by either the authorization plugin to clients or the subscription endpoint.
```

Comment:
@Geal I think I need more crispness on what you had meant when explaining this.  My **nuanced** take on what you had wrote was that this applied to ***subscription callback URL*** — which is _distinctly_ different from what Router clients would send to the Router.  In other words, two separate things:

1. When a client (e.g., Android, iOS) is making a request to the Route, if there is not a satisfactory answer from the JWT negotiation, then a 401 will be returned.
2. When using the subscription callback URL (i.e., when calling it from a subgraph implementation or pub/sub implementation; **not** a client-sourced request) if a:
   - Non-existent subscription identifier is used.
   - (I'm actually not sure what the difference is between an invalid hash and non-existent subscription — either way it's non-existent, I think?)

Reply:
on 1. if there is an issue when verifying the JWT, or if no JWT is present and `require_authentication` is set to true, then a 401 is returned
on 2. yes, it is about the subscription callback URL. It can give the subgraph a key that is used to hash the subscription event. If the hash does not match the event, the router will return a 401 to the callback request. And there's another case in heartbeats where the router wil return a 401 if the id is not in the expected list, but I am not clear on that, maybe @bnjjj could give more context (or I'll dig in the code)